File: src/store/mediaSlice.ts
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { MediaAsset } from '../types';

interface MediaState {
  items: MediaAsset[];
  isLoading: boolean;
  error: string | null;
  selectedItem: MediaAsset | null;
}

const initialState: MediaState = {
  items: [],
  isLoading: false,
  error: null,
  selectedItem: null,
};

const mediaSlice = createSlice({
  name: 'media',
  initialState,
  reducers: {
    setItems: (state: MediaState, action: PayloadAction<MediaAsset[]>) => {
      state.items = action.payload;
    },
    setLoading: (state: MediaState, action: PayloadAction<boolean>) => {
      state.isLoading = action.payload;
    },
    setError: (state: MediaState, action: PayloadAction<string | null>) => {
      state.error = action.payload;
    },
    setSelectedItem: (state: MediaState, action: PayloadAction<MediaAsset | null>) => {
      state.selectedItem = action.payload;
    },
  },
});

export const { setItems, setLoading, setError, setSelectedItem } = mediaSlice.actions;
export const mediaReducer = mediaSlice.reducer; 

File: backend/app/src/store/mediaSlice.py
from typing import Any, Dict, List, Union


class MediaState:
    items: List[MediaAsset]
    isLoading: bool
    error: Union[str, None]
    selectedItem: Union[MediaAsset, None]
const initialState: \'MediaState\' = {
  items: [],
  isLoading: false,
  error: null,
  selectedItem: null,
}
const mediaSlice = createSlice({
  name: 'media',
  initialState,
  reducers: Dict[str, Any],
    setLoading: (state: \'MediaState\', action: PayloadAction<boolean>) => {
      state.isLoading = action.payload
    },
    setError: (state: \'MediaState\', action: PayloadAction<string | null>) => {
      state.error = action.payload
    },
    setSelectedItem: (state: \'MediaState\', action: PayloadAction<MediaAsset | null>) => {
      state.selectedItem = action.payload
    },
  },
})
const { setItems, setLoading, setError, setSelectedItem } = mediaSlice.actions
const mediaReducer = mediaSlice.reducer 